### PR TITLE
Add test resolve from parent pom

### PR DESCRIPTION
**Pull request by ohonchar**
Moving selenium tests specifically ArtifactRepositoryBugfixTest. This is QE initiative to reduce selenium test suite.
Those tests trying to resolve missing GAV parameters from parent pom

### CODE REVIEW

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/MockFileItem.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    MockFileItem(final String fileName,
+                 final InputStream fileStream) {
+        this.fileName = fileName;
+        this.fileStream = fileStream;
+    }
+
+    @Override
+    public InputStream getInputStream() throws IOException {
+        return fileStream;
+        //return this.getClass().getResourceAsStream( "guvnor-m2repo-editor-backend-test-without-pom.jar" );
```
Comment (tomasdavidorg) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/MockFileItem.java]: I think this line can be deleted.

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import org.slf4j.LoggerFactory;
+import org.uberfire.backend.server.cdi.workspace.WorkspaceNameResolver;
+import org.uberfire.mocks.MockInstanceImpl;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class M2RepoServiceCreator {
+
+    private static final Logger log = LoggerFactory.getLogger(M2MavenRepositoryServiceImplTest.class);
```
Comment (tomasdavidorg) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java]: Should be `M2RepoServiceCreator.class`.

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    GuvnorM2Repository repo;
```
Comment (tomasdavidorg) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java]: I would rather add getters for these variables.

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GAVResolverFromParentPomTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.guvnor.m2repo.backend.server.M2RepoServiceCreator.deleteDir;
+import static org.guvnor.m2repo.model.HTMLFileManagerFields.UPLOAD_OK;
+
+public class GAVResolverFromParentPomTest {
+
+    private static final Logger log = LoggerFactory.getLogger(M2MavenRepositoryServiceImplTest.class);
```
Comment (tomasdavidorg) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GAVResolverFromParentPomTest.java]: Should be `GAVResolverFromParentPomTest.class`

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GAVResolverFromParentPomTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    private static final String parentPom = "parent-gav-pom.xml";
```
Comment (tomasdavidorg) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/GAVResolverFromParentPomTest.java]: Use uppercase letters.

File: uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        helperMethod.setAccessible(true);
+
+        //Set the repository service created above in the HttpPostHelper
+        java.lang.reflect.Field m2RepoServiceField = HttpPostHelper.class.getDeclaredField("m2RepoService");
+        m2RepoServiceField.setAccessible(true);
+        m2RepoServiceField.set(helper,
+                               service);
+    }
+
+    static boolean deleteDir(File dir) {
```
Comment (jhrcek) [inline uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2RepoServiceCreator.java]: I'd recommend using something like `FileUtils.deleteDirectory(f);` from commons-io (it's already used in appformer, you can add it as test dependency to this module)

**Comment by ohonchar** [comment]
@ederign , @tomasdavidorg , @jhrcek All changes have done. Could u please review once again. Thnx.

**Pull request merged**